drape_frontend/apply_feature_functors.cpp

```diff
@@ -486,18 +498,38 @@ void ApplyPointFeature::Finish(ref_ptr<dp::TextureManager> texMng)
       LOG(LERROR, ("Style error. Symbol name must be valid for feature", m_id));
   }
 
-  for (auto textParams : m_textParams)
+  bool const hasText = !m_textParams.m_titleDecl.m_primaryText.empty();
+  bool const hasHouseNumber = !m_hnParams.m_titleDecl.m_primaryText.empty();
+  if (hasText)
   {
     /// @todo Hardcoded styles-bug patch. The patch is ok, but probably should enhance (or fire assert) styles?
     /// @see https://github.com/organicmaps/organicmaps/issues/2573
-    if (hasPOI && textParams.m_titleDecl.m_anchor == dp::Anchor::Center)
+    if ((hasIcon || hasHouseNumber) && m_textParams.m_titleDecl.m_anchor == dp::Anchor::Center)
     {
-      textParams.m_titleDecl.m_anchor = GetAnchor(0, 1);
-      textParams.m_titleDecl.m_primaryOffset = GetOffset(0, 1);
+      ASSERT(!hasIcon, ("A `text-offset: *` is not set in styles.", m_id, m_textParams.m_titleDecl.m_primaryText));
```

Comment:
Ok, rewrite like:
```
if (m_textParams.m_titleDecl.m_anchor == dp::Anchor::Center)
{
  ASSERT(!hasIcon, (m_id));
  if (hasHouseNumber)
    m_textParams.m_titleDecl.m_anchor = GetAnchor(0, 1);
}
```

Reply:
Still, we'd want to fix the anchor for the icon case, so the second `if` would be `if (hasHouseNumber || hasIcon)`

Reply:
So I don't know if this code will be any better, conditions will be the same, but just split into two lines..

Reply:
> Still, we'd want to fix the anchor for the icon case, so the second if would be if (hasHouseNumber || hasIcon)
No because ASSERT(!hasIcon). Or I incorrectly interpret your code?

I try to avoid misambiguity. We have:
if (X || ...)
{
  ASSERT(!X, ());  // so if we have !X here, why do we check if (X) above?
 }

Reply:
There could be cases in prod when there is icon and anchor is set to center.
We want to fix them at runtime and at the same time fire an assert in debug to catch those cases and fix in styles.